tests/PidTest.cpp

```diff
@@ -0,0 +1,331 @@
+    double currentValue = 20;
+    /** 計算過程
+     * 1. 前回の誤差
+     * prevDeviation = 0
+     * 2. 現在の誤差
+     * currentDeviation = (70 - 20) = 50
+     * 3. 誤差の積分を計算
+     * integral = 0 + (50 + 0) * 0.01 / 2 = 0.25
+     * 4. 微分の処理を行う
+     * currentDerivative = (50 - 0) / 0.01 = 5000
+     * 微分項にローパスフィルタを適用
+     * prevDeviat filteredDerivative = 0.8 * 5000 + (1 - 0.8) * 0 = 4000
+     * 5. PID制御を計算
+     * expected = 0.6 * 50 + 0.02 * 0.25 + 0.03 * 4000 = 150.005
+     */
+    double expected = 150.005;
+    EXPECT_DOUBLE_EQ(expected, actualPid.calculatePid(currentValue));
+  }
+
+  // ゲインが全て0の時のPID計算の結果が正しいかをテスト
+  TEST(PidTest, CalculatePidGainZero)
+  {
+    Pid actualPid(0.0, 0.0, 0.0, 70, 100.0, -100.0);
+    double currentValue = 20;
+    /** 計算過程
+     * 1. 前回の誤差
+     * prevDeviation = 0
+     * 2. 現在の誤差
+     * currentDeviation = (70 - 20) = 50
+     * 3. 誤差の積分を計算
+     * integral = 0 + (50 + 0) * 0.01 / 2 = 0.25
+     * 4. 微分の処理を行う
+     * currentDerivative = (50 - 0) / 0.01 = 5000
+     * 微分項にローパスフィルタを適用
+     * prevDeviat filteredDerivative = 0.8 * 5000 + (1 - 0.8) * 0 = 4000
+     * 5. PID制御を計算
+     * expected = 0.0 * 50 + 0.0 * 0.25 + 0.0 * 4000 = 0.0
+     */
+    double expected = 0.0;
+    EXPECT_DOUBLE_EQ(expected, actualPid.calculatePid(currentValue));
+  }
+
+  // 周期を変更した場合のPID計算結果をテスト
+  TEST(PidTest, CalculatePidChangeDelta)
+  {
+    double DELTA = 0.03;
+    Pid actualPid(0.6, 0.02, 0.03, 70, 100.0, -100.0);
+    double currentValue = 55;
+    /** 計算過程
+     * 1. 前回の誤差
+     * prevDeviation = 0
+     * 2. 現在の誤差
+     * currentDeviation = (70 - 55) = 15
+     * 3. 誤差の積分を計算
+     * integral = 0 + (15 + 0) * 0.03 / 2 = 0.225
+     * 4. 微分の処理を行う
+     * currentDerivative = (15 - 0) / 0.03 = 500
+     * 微分項にローパスフィルタを適用
+     * filteredDerivative = 0.8 * 500 + (1 - 0.8) * 0 = 400
+     * 5. PID制御を計算
+     * expected = 0.6 * 15 + 0.02 * 0.225 + 0.03 * 400 = 21.0045
+     */
+    double expected = 21.0045;
+    EXPECT_DOUBLE_EQ(expected, actualPid.calculatePid(currentValue, DELTA));
+  }
+
+  // 周期に0を渡したとき、内部でデフォルトの0.01が使用されるかをテスト
+  TEST(PidTest, CalculatePidChangeDeltaZero)
+  {
+    Pid actualPid(0.6, 0.02, 0.03, 70, 100.0, -100.0);
+    double currentValue = 55;
+    /** 計算過程
+     * 1. 前回の誤差
+     * prevDeviation = 0
+     * 2. 現在の誤差
+     * currentDeviation = (70 - 55) = 15
+     * 3. 誤差の積分を計算
+     * integral = 0 + (15 + 0) * 0.01 / 2 = 0.075
+     * 4. 微分の処理を行う
+     * currentDerivative = (15 - 0) / 0.01 = 1500
+     * 微分項にローパスフィルタを適用
+     * filteredDerivative = 0.8 * 1500 + (1 - 0.8) * 0 = 1200
+     * 5. PID制御を計算
+     * expected = 0.6 * 15 + 0.02 * 0.075 + 0.03 * 1200 = 45.0015
+     */
+    double expected = 45.0015;
+    EXPECT_DOUBLE_EQ(expected, actualPid.calculatePid(currentValue, 0.0));
+  }
+
+  // setPidGainでゲイン変更後にPID計算が新しいゲインを使用しているかをテスト
+  TEST(PidTest, CalculatePidAfterSetNewGain)
+  {
+    Pid actualPid(0.6, 0.05, 0.01, 70, 100.0, -100.0);
+    double preCurrentValue = 60;
```

Comment:
```suggestion
    double prevValue = 60;
```
さすがにここは変えた方がいい気がする。前今になってるから。
あと、prevDeviation みたいに、前は prev で命名してるからそれに合わせた方がいいかな。
他も同様。